src/pkg/exporter/js_collector.go

```diff
@@ -0,0 +1,103 @@
+package exporter
+
+import (
+	"github.com/gomodule/redigo/redis"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// JobServiceCollectorName ...
+const JobServiceCollectorName = "JobServiceCollector"
+
+var (
+	jobServiceTaskQueueSize = typedDesc{
+		desc:      newDescWithLables("", "task_queue_size", "Total number of tasks", "type"),
+		valueType: prometheus.GaugeValue,
+	}
+	jobServiceTaskQueueLatency = typedDesc{
+		desc:      newDescWithLables("", "task_queue_latency", "how long ago the next job to be processed was enqueued", "type"),
+		valueType: prometheus.GaugeValue,
+	}
+	jobServiceConcurrency = typedDesc{
+		desc:      newDescWithLables("", "task_concurrency", "Total number of concurrency on a pool", "type", "pool"),
+		valueType: prometheus.GaugeValue,
+	}
+	jobServiceScheduledJobTotal = typedDesc{
+		desc:      newDesc("", "task_scheduled_total", "total number of scheduled job"),
+		valueType: prometheus.GaugeValue,
+	}
+	jobServiceScheduledJobFails = typedDesc{
+		desc:      newDescWithLables("", "task_scheduled_fails", "fail number of a scheduled job", "type"),
+		valueType: prometheus.GaugeValue,
+	}
+)
+
+// NewJobServiceCollector ...
+func NewJobServiceCollector() *JobServiceCollector {
+	return &JobServiceCollector{Namespace: namespace}
+}
+
+// JobServiceCollector ...
+type JobServiceCollector struct {
+	Namespace string
+}
+
+// Describe implements prometheus.Collector
+func (hc *JobServiceCollector) Describe(c chan<- *prometheus.Desc) {
+	c <- jobServiceTaskQueueSize.Desc()
```

Comment:
Are all the `Desc()` returning `typedDesc`? If yes, why not make the logic simpler?

```go
d:=[]typedDesc{
  jobServiceTaskQueueSize.Desc(),
  jobServiceTaskQueueLatency.Desc(),
  jobServiceConcurrency.Desc(),
  jobServiceScheduledJobFails.Desc(),
}

for _, jd := range d {
   c <- jd
}

```